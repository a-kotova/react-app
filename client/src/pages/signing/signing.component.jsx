import React from 'react';
import { createStructuredSelector } from 'reselect';
import { connect } from 'react-redux';
import { SigningPageContainer, ErrorContainer, SigningPageFormsContainer} from './signing.styles';
import SignIn from '../../components/sign-in/sign-in.component';
import SignUp from '../../components/sign-up/sign-up.component';
import { selectError } from '../../redux/user/user.selectors';

export const SigningPage = ({ error }) => (
  <SigningPageContainer>
    {
      (error && error.message !== 'snapshot.data is not a function')
        ? <ErrorContainer data-test="signing-error">{error.message}</ErrorContainer>
        : null
    }
    <SigningPageFormsContainer>
      <SignIn />
      <SignUp />
    </SigningPageFormsContainer>
  </SigningPageContainer>
);

export const mapStateToProps = createStructuredSelector({
  error: selectError,
});

export default connect(mapStateToProps)(SigningPage);
