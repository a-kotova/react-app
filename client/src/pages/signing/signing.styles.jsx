import styled from 'styled-components';

const SigningPageContainer = styled.div`
  width: 850px;
  display: flex;
  justify-content: space-between;
  margin: 30px auto;
`;

export default SigningPageContainer;
