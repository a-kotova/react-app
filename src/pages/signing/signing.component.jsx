import React from "react";
import "./signing.styles.scss"
import SignIn from "../../components/sign-in/sign-in.component";

const SigningPage = () => {
  return (
    <div>
      <h2>SIGN PAGE</h2>
      <SignIn />
    </div>
  )
}

export default SigningPage
