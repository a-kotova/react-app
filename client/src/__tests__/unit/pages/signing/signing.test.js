import React from 'react';
import { shallow } from 'enzyme';
import { SigningPage } from '../../../../pages/signing/signing.component';

describe('Pages: Signing Page', () => {
  it('should render without errors', () => {
    const wrapper = shallow(<SigningPage />);

    expect(wrapper).toMatchSnapshot();
  });
});
