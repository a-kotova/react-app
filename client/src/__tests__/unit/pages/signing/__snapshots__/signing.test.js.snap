// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages: Signing Page should render without errors 1`] = `
<SigningPageContainer>
  <Connect(SignIn) />
  <Connect(SignUp) />
</SigningPageContainer>
`;
